pkg/scheduler/plugins/elasticquota/pod_handler.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package elasticquota
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+func (g *Plugin) OnPodAdd(obj interface{}) {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return
+	}
+	g.Lock()
+	defer g.Unlock()
+
+	quotaName := g.getPodAssociateQuotaName(pod)
+	g.groupQuotaManager.UpdatePodCache(quotaName, pod, true)
+	g.groupQuotaManager.UpdatePodRequest(quotaName, nil, pod)
+	// in case failOver, update pod isAssigned explicitly according to its phase.
+	if pod.Status.Phase == corev1.PodRunning {
+		g.groupQuotaManager.UpdatePodIsAssigned(quotaName, pod, true)
+		g.groupQuotaManager.UpdatePodUsed(quotaName, nil, pod)
+	}
+	klog.V(5).Infof("OnPodAddFunc %v.%v add success, quotaName:%v", pod.Namespace, pod.Name, quotaName)
+}
+
+func (g *Plugin) OnPodUpdate(oldObj, newObj interface{}) {
+	oldPod := oldObj.(*corev1.Pod)
+	newPod := newObj.(*corev1.Pod)
+
+	g.Lock()
+	defer g.Unlock()
+
```

Comment:
We do the following operations in the handler. If the operation gets stuck (for some reason), it will block the event processing of the entire informer. So the following operation may be stuck for a long time?

Reply:
fixed, delete the rwlock from plugin.